test/e2e/configuration_test.go

```diff
@@ -70,5 +75,113 @@ func TestConfiguration(t *testing.T) {
 			}
 		}
 	}
+}
+
+func TestRouteConfiguration(t *testing.T) {
+	client := testframework.MustNewClientset(t, nil)
 
+	defer testframework.MustRemoveImageRegistry(t, client)
+
+	cr := &imageregistryv1.Config{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: imageregistryv1.SchemeGroupVersion.String(),
+			Kind:       "Config",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: imageregistryv1.ImageRegistryResourceName,
+		},
+		Spec: imageregistryv1.ImageRegistrySpec{
+			ManagementState: operatorapiv1.Managed,
+			Storage: imageregistryv1.ImageRegistryConfigStorage{
+				Filesystem: &imageregistryv1.ImageRegistryConfigStorageFilesystem{
+					VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+			Replicas:     1,
+			DefaultRoute: true,
+			Routes: []imageregistryv1.ImageRegistryConfigRoute{
+				{
+					Name:     "testroute",
+					Hostname: "test.example.com",
+				},
+			},
+		},
+	}
+	testframework.MustDeployImageRegistry(t, client, cr)
+	testframework.MustEnsureImageRegistryIsAvailable(t, client)
+	testframework.MustEnsureClusterOperatorStatusIsSet(t, client)
+	ensureExternalRegistryHostnamesAreSet(t, client)
+	ensureExternalRoutesExist(t, client)
+
+}
+
+func ensureExternalRegistryHostnamesAreSet(t *testing.T, client *testframework.Clientset) {
+	var cfg *configapiv1.Image
+	var err error
+	externalHosts := []string{}
+	err = wait.Poll(1*time.Second, testframework.AsyncOperationTimeout, func() (bool, error) {
+		var err error
+		cfg, err = client.Images().Get("cluster", metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			t.Logf("waiting for the image config resource: the resource does not exist")
+			cfg = nil
+			return false, nil
+		} else if err != nil {
+			return false, err
+		}
+		if cfg == nil {
+			return false, nil
+		}
+		externalHosts = cfg.Status.ExternalRegistryHostnames
+
+		foundDefaultRoute := false
+		foundUserRoute := false
+		for _, h := range externalHosts {
+			if strings.HasPrefix(h, "default-route-openshift-image-registry") {
```

Comment:
Can we build default-route-openshift-image-registry so that this test doesn't have to be updated if we change that name?

Reply:
updated